lib/oban/config.ex

```diff
@@ -140,7 +140,8 @@ defmodule Oban.Config do
     Validation.validate_schema(opts,
       dispatch_cooldown: :pos_integer,
       engine: {:behaviour, Oban.Engine},
-      get_dynamic_repo: {:or, [:falsy, {:function, 0}]},
+      get_dynamic_repo:
+        {:or, [:falsy, {:function, 0}, {:custom, &validate_mfa(:get_dynamic_repo, &1)}]},
```

Comment:
If theres a better way to do this I'm all ears

Reply:
How about adding `mfa` to the validator?